# Top Earners in Movie Industry

## Table of Contents

<ul>
    <li><a href="#intro">Introduction</a></li>
    <li><a href="#eda">Exploratory Data Analysis</a></li>
    <li><a href="#conclusion">Conclusion</a></li>
</ul>

<a id="#intro"></a>
## Introduction

> This analysis project is to be done using the imdb movie data. When the analysis is completed, you should be able to find the top 5 highest grossing directors, the top 5 highest grossing movie genres of all time, comparing the revenue of the highest grossing movies and which companies released the most movies. 

> There are 10 columns that will not be needed for the analysis. Use pandas to drop these columns. HINT: Only the columns pertaining to revenue will be needed.

> To get you started, I've already placed the needed code for getting the packages and datafile that you will be using for the project. 

In [112]:
import pandas as pd
import numpy as np

%matplotlib inline
import matplotlib.pyplot as plt

In [113]:
df = pd.read_csv('files\imdb-movies.csv')
df.head(3)
#revenue = df['revenue']
#revenue.head()

id    imdb_id  popularity     budget     revenue      original_title  \
0  135397  tt0369610   32.985763  150000000  1513528810      Jurassic World   
1   76341  tt1392190   28.419936  150000000   378436354  Mad Max: Fury Road   
2  262500  tt2908446   13.112507  110000000   295238201           Insurgent   

                                                cast  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2  Shailene Woodley|Theo James|Kate Winslet|Ansel...   

                                         homepage          director  \
0                   http://www.jurassicworld.com/   Colin Trevorrow   
1                     http://www.madmaxmovie.com/     George Miller   
2  http://www.thedivergentseries.movie/#insurgent  Robert Schwentke   

                      tagline  ...  \
0           The park is open.  ...   
1          What a Lovely Day.  ...   
2  One Choice Can Destroy You  ...   

                                            overview runtime  \
0  Twenty-two years after the events of Jurassic ...     124   
1  An apocalyptic story set in the furthest reach...     120   
2  Beatrice Prior must confront her inner demons ...     119   

                                      genres  \
0  Action|Adventure|Science Fiction|Thriller   
1  Action|Adventure|Science Fiction|Thriller   
2         Adventure|Science Fiction|Thriller   

                                production_companies release_date vote_count  \
0  Universal Studios|Amblin Entertainment|Legenda...       6/9/15       5562   
1  Village Roadshow Pictures|Kennedy Miller Produ...      5/13/15       6185   
2  Summit Entertainment|Mandeville Films|Red Wago...      3/18/15       2480   

   vote_average  release_year    budget_adj   revenue_adj  
0           6.5          2015  1.379999e+08  1.392446e+09  
1           7.1          2015  1.379999e+08  3.481613e+08  
2           6.3          2015  1.012000e+08  2.716190e+08  

[3 rows x 21 columns]

### Drop columns without neccesary information and remove all records with no financial information -- Pay close attention to things that don't tell you anything regarding financial data

In [114]:
df.drop(df.query('budget == 0 or revenue == 0').index, inplace = True)
df.drop(['id','popularity','imdb_id','homepage','budget', 'runtime','tagline','keywords','vote_count','vote_average','release_date','budget_adj','overview'], axis=1, inplace=True)

In [115]:
df.dropna(inplace = True)
df.reset_index(drop = True, inplace = True)
df.head()

revenue                original_title  \
0  1513528810                Jurassic World   
1   378436354            Mad Max: Fury Road   
2   295238201                     Insurgent   
3  2068178225  Star Wars: The Force Awakens   
4  1506249360                     Furious 7   

                                                cast          director  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   Colin Trevorrow   
1  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...     George Miller   
2  Shailene Woodley|Theo James|Kate Winslet|Ansel...  Robert Schwentke   
3  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...       J.J. Abrams   
4  Vin Diesel|Paul Walker|Jason Statham|Michelle ...         James Wan   

                                      genres  \
0  Action|Adventure|Science Fiction|Thriller   
1  Action|Adventure|Science Fiction|Thriller   
2         Adventure|Science Fiction|Thriller   
3   Action|Adventure|Science Fiction|Fantasy   
4                      Action|Crime|Thriller   

                                production_companies  release_year  \
0  Universal Studios|Amblin Entertainment|Legenda...          2015   
1  Village Roadshow Pictures|Kennedy Miller Produ...          2015   
2  Summit Entertainment|Mandeville Films|Red Wago...          2015   
3          Lucasfilm|Truenorth Productions|Bad Robot          2015   
4  Universal Pictures|Original Film|Media Rights ...          2015   

    revenue_adj  
0  1.392446e+09  
1  3.481613e+08  
2  2.716190e+08  
3  1.902723e+09  
4  1.385749e+09

In [116]:
production_records = df[df.production_companies.str.contains('|')]
production_records.production_companies = production_records.production_companies.apply(lambda x: x.split('|'))
df_pc = pd.DataFrame(columns = df.columns)

for i in range(len(production_records)):
    record = production_records.iloc[i] # returns all data for every single record with the ability to access columns
    for production_company in production_records.production_companies[i]: # loop through ALL production_companies from EACH record
        df_pc = df_pc.append(pd.DataFrame([[record.revenue, record.original_title, record.cast, record.director,record.genres, production_company, record.release_year, record.revenue_adj]], columns = df.columns))
            
df_pc.drop('genres', axis=1, inplace=True)
df_pc.reset_index(drop=True,inplace=True)

In [117]:
df_pc.head()

revenue  original_title  \
0  1513528810  Jurassic World   
1  1513528810  Jurassic World   
2  1513528810  Jurassic World   
3  1513528810  Jurassic World   
4  1513528810  Jurassic World   

                                                cast         director  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
1  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
2  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
3  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
4  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   

      production_companies release_year   revenue_adj  
0        Universal Studios         2015  1.392446e+09  
1     Amblin Entertainment         2015  1.392446e+09  
2       Legendary Pictures         2015  1.392446e+09  
3  Fuji Television Network         2015  1.392446e+09  
4                   Dentsu         2015  1.392446e+09

In [118]:
genere_type = df[df.genres.str.contains('|')]
genere_type.genres = genere_type.genres.apply(lambda x: x.split('|'))
df_gt = pd.DataFrame(columns = df.columns)

for i in range(len(genere_type)):
    record = genere_type.iloc[i] # returns all data for every single record with the ability to access columns
    for genre in genere_type.genres[i]: # loop through ALL production_companies from EACH record
        df_gt = df_gt.append(pd.DataFrame([[record.revenue, record.original_title, record.cast, record.director, genre, record.production_companies, record.release_year, record.revenue_adj]], columns = df.columns))
            
df_gt.drop('production_companies', axis=1, inplace=True)
df_gt.reset_index(drop=True,inplace=True)

In [119]:
df_gt.head()

revenue      original_title  \
0  1513528810      Jurassic World   
1  1513528810      Jurassic World   
2  1513528810      Jurassic World   
3  1513528810      Jurassic World   
4   378436354  Mad Max: Fury Road   

                                                cast         director  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
1  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
2  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
3  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
4  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...    George Miller   

            genres release_year   revenue_adj  
0           Action         2015  1.392446e+09  
1        Adventure         2015  1.392446e+09  
2  Science Fiction         2015  1.392446e+09  
3         Thriller         2015  1.392446e+09  
4           Action         2015  3.481613e+08

### Data Cleaning

In [74]:
# Delete all records with null, or empty values

# df.dropna(inplace=True)
# df

revenue                original_title  \
0      1513528810                Jurassic World   
1       378436354            Mad Max: Fury Road   
2       295238201                     Insurgent   
3      2068178225  Star Wars: The Force Awakens   
4      1506249360                     Furious 7   
...           ...                           ...   
10861           0            The Endless Summer   
10862           0                    Grand Prix   
10863           0           Beregis Avtomobilya   
10864           0        What's Up, Tiger Lily?   
10865           0      Manos: The Hands of Fate   

                                                    cast            director  \
0      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...     Colin Trevorrow   
1      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...       George Miller   
2      Shailene Woodley|Theo James|Kate Winslet|Ansel...    Robert Schwentke   
3      Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...         J.J. Abrams   
4      Vin Diesel|Paul Walker|Jason Statham|Michelle ...           James Wan   
...                                                  ...                 ...   
10861  Michael Hynson|Robert August|Lord 'Tally Ho' B...         Bruce Brown   
10862  James Garner|Eva Marie Saint|Yves Montand|Tosh...  John Frankenheimer   
10863  Innokentiy Smoktunovskiy|Oleg Efremov|Georgi Z...      Eldar Ryazanov   
10864  Tatsuya Mihashi|Akiko Wakabayashi|Mie Hama|Joh...         Woody Allen   
10865  Harold P. Warren|Tom Neyman|John Reynolds|Dian...    Harold P. Warren   

                                          genres  \
0      Action|Adventure|Science Fiction|Thriller   
1      Action|Adventure|Science Fiction|Thriller   
2             Adventure|Science Fiction|Thriller   
3       Action|Adventure|Science Fiction|Fantasy   
4                          Action|Crime|Thriller   
...                                          ...   
10861                                Documentary   
10862                     Action|Adventure|Drama   
10863                             Mystery|Comedy   
10864                              Action|Comedy   
10865                                     Horror   

                                    production_companies  release_year  \
0      Universal Studios|Amblin Entertainment|Legenda...          2015   
1      Village Roadshow Pictures|Kennedy Miller Produ...          2015   
2      Summit Entertainment|Mandeville Films|Red Wago...          2015   
3              Lucasfilm|Truenorth Productions|Bad Robot          2015   
4      Universal Pictures|Original Film|Media Rights ...          2015   
...                                                  ...           ...   
10861                                  Bruce Brown Films          1966   
10862  Cherokee Productions|Joel Productions|Douglas ...          1966   
10863                                            Mosfilm          1966   
10864                            Benedict Pictures Corp.          1966   
10865                                          Norm-Iris          1966   

        revenue_adj  
0      1.392446e+09  
1      3.481613e+08  
2      2.716190e+08  
3      1.902723e+09  
4      1.385749e+09  
...             ...  
10861  0.000000e+00  
10862  0.000000e+00  
10863  0.000000e+00  
10864  0.000000e+00  
10865  0.000000e+00  

[9773 rows x 8 columns]

#### Here's a helpful hint from my own analysis when I ran this the first time. This may help shed light on what your data set should look like.

#### If I created one record for each the `production_companies` a movie was release under and one record each for `genres`<br>and tried to run calculations, it wouldn't work because for many records, the amount of `production_companies`<br>and `genres` aren't the same, so I'll create 2 dataframes; one w/o a `production_companies` column and one w/o a `genres` columns

In [45]:
# df_genres = df.drop('production_companies', axis=1, inplace=False)
# df_production_companies = df.drop(['genres'], axis=1, inplace=False)

In [47]:
# df_genres.head()

revenue                original_title  \
0  1513528810                Jurassic World   
1   378436354            Mad Max: Fury Road   
2   295238201                     Insurgent   
3  2068178225  Star Wars: The Force Awakens   
4  1506249360                     Furious 7   

                                                cast          director  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   Colin Trevorrow   
1  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...     George Miller   
2  Shailene Woodley|Theo James|Kate Winslet|Ansel...  Robert Schwentke   
3  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...       J.J. Abrams   
4  Vin Diesel|Paul Walker|Jason Statham|Michelle ...         James Wan   

                                      genres  release_year   revenue_adj  
0  Action|Adventure|Science Fiction|Thriller          2015  1.392446e+09  
1  Action|Adventure|Science Fiction|Thriller          2015  3.481613e+08  
2         Adventure|Science Fiction|Thriller          2015  2.716190e+08  
3   Action|Adventure|Science Fiction|Fantasy          2015  1.902723e+09  
4                      Action|Crime|Thriller          2015  1.385749e+09

In [48]:
# df_production_companies.head()

revenue                original_title  \
0  1513528810                Jurassic World   
1   378436354            Mad Max: Fury Road   
2   295238201                     Insurgent   
3  2068178225  Star Wars: The Force Awakens   
4  1506249360                     Furious 7   

                                                cast          director  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   Colin Trevorrow   
1  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...     George Miller   
2  Shailene Woodley|Theo James|Kate Winslet|Ansel...  Robert Schwentke   
3  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...       J.J. Abrams   
4  Vin Diesel|Paul Walker|Jason Statham|Michelle ...         James Wan   

                                production_companies  release_year  \
0  Universal Studios|Amblin Entertainment|Legenda...          2015   
1  Village Roadshow Pictures|Kennedy Miller Produ...          2015   
2  Summit Entertainment|Mandeville Films|Red Wago...          2015   
3          Lucasfilm|Truenorth Productions|Bad Robot          2015   
4  Universal Pictures|Original Film|Media Rights ...          2015   

    revenue_adj  
0  1.392446e+09  
1  3.481613e+08  
2  2.716190e+08  
3  1.902723e+09  
4  1.385749e+09

<a id="eda"></a>
## Exploratory Data Analysis

> Use Matplotlib to display your data analysis

### Which production companies released the most movies in the last 10 years? Display the top 5 production companies.

In [120]:
# df_production_companies.groupby('production_companies', as_index = False)
# most_reased = df_production_companies.apply(lambda x: x.sort_values([""]))

last_10 = df_pc.query('release_year > 2007').production_companies.value_counts().nlargest(10)
last_10


Relativity Media                          90
Universal Pictures                        89
Warner Bros.                              87
Columbia Pictures                         73
Paramount Pictures                        60
Twentieth Century Fox Film Corporation    52
New Line Cinema                           45
Walt Disney Pictures                      41
Summit Entertainment                      38
Dune Entertainment                        37
Name: production_companies, dtype: int64

### What 5 movie genres grossed the highest all-time?

In [121]:
# df_genres.groupby('genres', as_index = False)
# highest_groseed = df_genres.apply(lambda x: x.sort_values([""]))
# highest_groseed.head(5)
# highest_groseed = df_genres.groupby('genres').apply(lambda x:x['revenue'].reset_index()).reset_index()
highest_groseed = df_gt.groupby('genres')['revenue'].sum().nlargest(5)
highest_groseed.head()

genres
Action       169790679429
Adventure    163439947064
Comedy       132020867028
Drama        130323128041
Thriller     117587434010
Name: revenue, dtype: int64

### Who are the top 5 grossing directors?

In [122]:
highest_groseed_dir = df_gt.groupby('director')['revenue'].sum().nlargest(5)
highest_groseed_dir.head()

director
Steven Spielberg     24663086098
James Cameron        20132327500
Peter Jackson        17530037629
Michael Bay          16297866403
Christopher Nolan    16196885522
Name: revenue, dtype: int64

### Compare the revenue of the highest grossing movies of all time.

In [123]:
highest_groseed_title = df_gt.groupby('original_title')['revenue'].sum().nlargest(5)
highest_groseed_title

original_title
Avatar                          11126023388
Star Wars: The Force Awakens     8272712900
Jurassic World                   6054115240
Titanic                          5535102564
The Net                          5531398290
Name: revenue, dtype: int64

<a id="conclusions"></a>
## Conclusions

> Using the cell below, write a brief conclusion of what you have found from the anaylsis of the data. The Cell below will allow you to write plan text instead of code.

Based on the anlysis of the above Movie data base, here are my top findings:
    1. Relative Media Company produced the most movies between the year of 2007 to 2017, 90 movies. 
    2. As far as the movie Genre goes, Action movies generated the most overall revenue, $169,790,679,429
    3. Among the top revenue generating director, Steven Spielberg was on top generating $24,663,086,098
    4. Avatar was the highest grssing movie from the data base, total revenue of $11,126,023,388